sdk/src/wallet/account/operations/transaction/input_selection.rs

```diff
@@ -261,44 +260,3 @@ fn filter_inputs(
 
     Ok(available_outputs_signing_data)
 }
-
-// Returns if alias transition is a state transition with the provided outputs for a given input.
-pub(crate) fn alias_state_transition(
-    output_data: &OutputData,
-    outputs: &[Output],
-    burn: Option<&Burn>,
-) -> crate::wallet::Result<Option<AliasTransition>> {
-    Ok(if let Output::Alias(alias_input) = &output_data.output {
-        let alias_id = alias_input.alias_id_non_null(&output_data.output_id);
-        // Check if alias exists in the outputs and get the required transition type
-        outputs
-            .iter()
-            .find_map(|o| {
-                if let Output::Alias(alias_output) = o {
-                    if *alias_output.alias_id() == alias_id {
-                        if alias_output.state_index() == alias_input.state_index() {
-                            Some(Some(AliasTransition::Governance))
-                        } else {
-                            Some(Some(AliasTransition::State))
-                        }
-                    } else {
-                        None
-                    }
-                } else {
-                    None
-                }
-                // if not find in the outputs, the alias gets burned which is a governance transaction
-            })
-            .unwrap_or_else(|| {
-                burn.and_then(|burn| {
```

Comment:
I need to investigate if removing this is actually fine

Reply:
I think it would still be required if governor and state controller have different addresses, because the bip32 path is then different (now it will just assume `AliasTransition::Governance`)
So I guess to make it fail you would need to
1. create an alias output with two different addresses for governor and state controller
2. send a basic output to the alias address
3. sync with basic outputs for aliases in the sync options enabled
4. manually consume the basic output created in 2. by providing it to mandatory_inputs and send it to any other address on the output side, but don't provide the alias output in the outputs
I think this transaction should then fail with the invalid signature

Reply:
Yes
```Rust
use iota_sdk::{
    client::{
        constants::SHIMMER_COIN_TYPE,
        generate_mnemonic, request_funds_from_faucet,
        secret::{mnemonic::MnemonicSecretManager, SecretManager},
    },
    types::block::{
        address::{Address, AliasAddress},
        output::{
            unlock_condition::{GovernorAddressUnlockCondition, StateControllerAddressUnlockCondition},
            AliasId, AliasOutputBuilder, OutputId,
        },
    },
    wallet::{
        account::{AliasSyncOptions, SyncOptions, TransactionOptions},
        ClientOptions, Result, SendAmountParams, Wallet,
    },
};

#[tokio::main]
async fn main() -> Result<()> {
    let client_options = ClientOptions::new().with_node(&"https://api.testnet.shimmer.network")?;

    let secret_manager = MnemonicSecretManager::try_from_mnemonic(&generate_mnemonic()?)?;

    let wallet = Wallet::builder()
        .with_secret_manager(SecretManager::Mnemonic(secret_manager))
        .with_client_options(client_options)
        .with_coin_type(SHIMMER_COIN_TYPE)
        .finish()
        .await?;

    let account = wallet.create_account().finish().await?;
    let first_address = &account.addresses().await?[0];

    let response = request_funds_from_faucet(
        "https://faucet.testnet.shimmer.network/api/enqueue",
        &first_address.address().to_string(),
    )
    .await?;
    println!("{response}");
    tokio::time::sleep(std::time::Duration::from_secs(30)).await;

    let balance = account.sync(None).await?;
    println!("Balance: {balance:?}");

    let second_address = &account.generate_addresses(1, None).await?[0];

    let token_supply = account.client().get_token_supply().await?;

    let alias_output = AliasOutputBuilder::new_with_amount(1_000_000, AliasId::null())
        .add_unlock_condition(GovernorAddressUnlockCondition::new(**first_address.address()))
        .add_unlock_condition(StateControllerAddressUnlockCondition::new(**second_address.address()))
        .finish_output(token_supply)?;

    let transaction = account.send(vec![alias_output], None).await?;
    account
        .retry_transaction_until_included(&transaction.transaction_id, None, None)
        .await?;
    println!(
        "Alias mint transaction: https://explorer.shimmer.network/testnet/transaction/{}",
        transaction.transaction_id
    );

    let balance = account.sync(None).await?;
    println!("Balance: {balance:?}");

    let alias_id = balance.aliases().first().unwrap();

    let transaction = account
        .send_amount(
            vec![SendAmountParams::new(
                Address::Alias(AliasAddress::new(*alias_id)).to_bech32("rms"),
                1_000_000,
            )],
            None,
        )
        .await?;
    account
        .retry_transaction_until_included(&transaction.transaction_id, None, None)
        .await?;
    println!(
        "Transaction with output to alias address: https://explorer.shimmer.network/testnet/transaction/{}",
        transaction.transaction_id
    );

    let balance = account
        .sync(Some(SyncOptions {
            alias: AliasSyncOptions {
                basic_outputs: true,
                ..Default::default()
            },
            ..Default::default()
        }))
        .await?;
    println!("Balance: {balance:?}");

    let input = OutputId::new(transaction.transaction_id, 0)?;
    let transaction = account
        .send_amount(
            vec![SendAmountParams::new(
                Address::Alias(AliasAddress::new(*alias_id)).to_bech32("rms"),
                1_000_000,
            )],
            Some(TransactionOptions {
                mandatory_inputs: Some(vec![input]),
                ..Default::default()
            }),
        )
        .await?;
    account
        .retry_transaction_until_included(&transaction.transaction_id, None, None)
        .await?;
    println!(
        "Transaction with custom input: https://explorer.shimmer.network/testnet/transaction/{}",
        transaction.transaction_id
    );

    Ok(())
}
```
fails with `Error: Client(TransactionSemantic(InvalidSignature))`

Reply:
@Alexandcoats so please include the optional burn check

Reply:
My example works now, should be good to go :heavy_check_mark: